### DISCUSSION TITLE
Multiple config dirs

### DISCUSSION THREAD

**Discussion by brianjmurrell** [rpm-software-management/mock#1078] [Q&A] [answered]
Is it possible to tell mock to use multiple config dirs?  Say I wanted a custom configuration in a dir I am able to create and use `--configdir` to specify that dir but I then want mock to fall back to the default config dir in case there are `include()` statements in the custom config dir that refer to files in the default config dir such as https://github.com/rpm-software-management/mock/blob/f9e1260e711a18b2a337baf059f4d995486c8182/mock-core-configs/etc/mock/alma%2Bepel-9-x86_64.cfg#L1

**Comment by xsuchy** [comment, marked as answer]
No. You can have more config and they are read in specified order. See bottom of https://rpm-software-management.github.io/mock/configuration

But you cannot specify multiple configdirs. 

Note that config dir is just a python file, so you can do anything there. But do not depend too much on that as we have plan to change it. But I think you are safe for year or two. Or five.

**Comment by brianjmurrell** [comment]
But you have to admit that the practice of starting to include other configs in a config makes it difficult to just pull one config from the configs, place it into a new dir, modify it as you want and then refer to that new dir to get your modified config.

Now we have to copy the entire config dir rather than just having mock fall back to the original config dir to pick up the unmodified config files that are referenced in a modified config's `include()`.

Does the above methodology for modifying a stock config not make sense?  Is there a better way to achieve that?

**Reply by praiskup** [reply]
With the include statement in place, never touch --configdir (you simply work with a single custom file, depending on the system config dir unmodified). Can you elaborate more on the case you want to do/implement?

**Reply by brianjmurrell** [reply]
Our environment has it's own repository mirror/cache, therefore we copy the standard mock configuration (for a given platform) to our own dir (i.e. where we are allowed to write, because we are not root and cannot write into `/etc/mock`), add our own repo configurations to that copy and then call mock with the `--configdir` option to use that dir we created and modified a copy of the config file along with options to disable the standard repos and enable our own local repos.

In some cases we alter module configurations in our copy of the config.

If there is a better way of doing this, I am open to suggestions.

**Comment by xsuchy** [comment]
If you put your config files in  `~/.config/mock/`, Then they will basically have higher priority than the ones in `/etc/mock`.  
Technically both will be evaluated, but if your file in `~/.config/mock/` is identical with different values then it will overwrite the values set in `/etc/mock`. And you do not need to pass `--configdir`.

**Comment by brianjmurrell** [comment]
Indeed.  It's even easier/better than that.  If you append `.cfg` to the `mock` `-r` argument it is taken as a path to a config file rather than a name to be used with `/etc/mock/` and/or `~/.config/mock/`.

This completely satisfies my requirement.
